homeassistant/components/template/config_flow.py

```diff
@@ -0,0 +1,320 @@
+    """Return next step_id for options flow according to template_type."""
+    return cast(str, options["template_type"])
+
+
+def _strip_sentinel(options: dict[str, Any]) -> None:
+    """Convert sentinel to None."""
+    for key in (CONF_DEVICE_CLASS, CONF_STATE_CLASS, CONF_UNIT_OF_MEASUREMENT):
+        if key not in options:
+            continue
+        if options[key] == NONE_SENTINEL:
+            options.pop(key)
+
+
+def _validate_unit(options: dict[str, Any]) -> None:
+    """Validate unit of measurement."""
+    if (
+        (device_class := options.get(CONF_DEVICE_CLASS))
+        and (units := DEVICE_CLASS_UNITS.get(device_class)) is not None
+        and (unit := options.get(CONF_UNIT_OF_MEASUREMENT)) not in units
+    ):
+        units_string = sorted(
+            [str(unit) if unit else "no unit of measurement" for unit in units],
+            key=str.casefold,
+        )
+
+        raise vol.Invalid(
+            f"'{unit}' is not a valid unit for device class '{device_class}'; "
+            f"expected one of {', '.join(units_string)}"
+        )
+
+
+def validate_user_input(
+    template_type: str,
+) -> Callable[
+    [SchemaCommonFlowHandler, dict[str, Any]],
+    Coroutine[Any, Any, dict[str, Any]],
+]:
+    """Do post validation of user input.
+
+    For sensors: Strip none-sentinels and validate unit of measurement.
+    For all domaines: Set template type.
+    """
+
+    async def _validate_user_input(
+        _: SchemaCommonFlowHandler,
+        user_input: dict[str, Any],
+    ) -> dict[str, Any]:
+        """Add template type to user input."""
+        if template_type == Platform.SENSOR:
+            _strip_sentinel(user_input)
+            _validate_unit(user_input)
+        return {"template_type": template_type} | user_input
+
+    return _validate_user_input
+
+
+TEMPLATE_TYPES = [
+    "sensor",
+]
+
+CONFIG_FLOW = {
+    "user": SchemaFlowMenuStep(TEMPLATE_TYPES),
+    Platform.SENSOR: SchemaFlowFormStep(
+        config_schema(Platform.SENSOR),
+        preview="template",
+        validate_user_input=validate_user_input(Platform.SENSOR),
+    ),
+}
+
+
+OPTIONS_FLOW = {
+    "init": SchemaFlowFormStep(next_step=choose_options_step),
+    Platform.SENSOR: SchemaFlowFormStep(
+        options_schema(Platform.SENSOR),
+        preview="template",
+        validate_user_input=validate_user_input(Platform.SENSOR),
+    ),
+}
+
+CREATE_PREVIEW_ENTITY: dict[
+    str,
+    Callable[[HomeAssistant, str, dict[str, Any]], TemplateEntity],
+] = {
+    "sensor": async_create_preview_sensor,
+}
+
+
+class TemplateConfigFlowHandler(SchemaConfigFlowHandler, domain=DOMAIN):
+    """Handle config flow for template helper."""
+
+    config_flow = CONFIG_FLOW
+    options_flow = OPTIONS_FLOW
+
+    @callback
+    def async_config_entry_title(self, options: Mapping[str, Any]) -> str:
+        """Return config entry title."""
+        return cast(str, options["name"]) if "name" in options else ""
```

Comment:
I don't think we should have an empty string as config entry title.

Reply:
The string is not optional, changed so the title is always the name entered by the user.